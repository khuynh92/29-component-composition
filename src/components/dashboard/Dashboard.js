import React, { Component, Fragment } from 'react';

import NoteCreateForm from '../note-create-form/NoteCreateForm.js';
import NoteList from '../note-list/NoteList.js';
import uuid from 'uuid/v4';

export default class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      content: '',
      title: '',
      notes: [],
    };
  }

  componentDidMount() {
    if (localStorage && localStorage.state) {
      let stateFromStorage = JSON.parse(localStorage.state);
      this.setState({ ...stateFromStorage });
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({
      [e.target.id]: e.target.value,
    });
  }

  addNote = (e) => {
    e.preventDefault();

    let newNote = {
      id: uuid(),
      title: this.state.title,
      content: this.state.content,
      editing: false,
      completed: false,
    };

    this.setState((prevState) => {
      return {
        content: '',
        title: '',
        notes: [...prevState.notes, newNote],
      };
    }, () => localStorage.state = JSON.stringify(this.state)
    );
  }

  removeNote = (note) => {

    let arr = [...this.state.notes];

    for (let i = 0; i < arr.length; i++) {
      if (arr[i].id === note.id) {
        arr.splice(i, 1);
        break;
      }
    }

    this.setState({
      notes: arr,
    }, () => localStorage.state = JSON.stringify(this.state)
    );
  };

  render() {
    return (
      <Fragment>
        <NoteCreateForm title={this.state.title} content={this.state.content} handleSubmit={this.addNote} handleChange={this.handleChange} />
        <NoteList removeNote={this.removeNote} notes={this.state.notes} />
      </Fragment>

    );
  }
}