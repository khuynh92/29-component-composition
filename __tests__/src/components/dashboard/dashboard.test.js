import React from 'react';
import renderer from 'react-test-renderer';

import Dashboard from '../../../../src/components/dashboard/Dashboard.js';

describe('<Dashboard/> Enzyme Test', () => {
  it('loads components at application start', () => {
    let app = shallow(<Dashboard/>);
    expect(app.find('NoteCreateForm').exists()).toBeTruthy();
    expect(app.find('NoteList').exists()).toBeTruthy();

  });
});

describe('<Dashboard/> Snapshot Test', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Dashboard />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});


