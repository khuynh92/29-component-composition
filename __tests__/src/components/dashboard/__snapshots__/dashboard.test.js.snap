// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> Snapshot Test renders correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label>
      Title:
      <br />
      <input
        id="title"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Notes:
      <br />
      <textarea
        id="content"
        onChange={[Function]}
        required={true}
        value=""
      />
    </label>
    <br />
    <input
      type="submit"
    />
  </form>,
  <ul
    id="note-list"
  />,
]
`;
